tests/neo.UnitTests/SmartContract/Native/UT_GasToken.cs

```diff
@@ -35,6 +36,67 @@ public void TestSetup()
         [TestMethod]
         public void Check_Decimals() => NativeContract.GAS.Decimals(_snapshot).Should().Be(8);
 
+        [TestMethod]
+        public void Refuel()
+        {
+            // Prepare
+
+            var wallet = TestUtils.GenerateTestWallet();
+            var snapshot = TestBlockchain.GetTestSnapshot();
+
+            using var unlock = wallet.Unlock("");
+            var accBalance = wallet.CreateAccount();
+            var accNoBalance = wallet.CreateAccount();
+
+            // Fake balance
+
+            var key = NativeContract.GAS.CreateStorageKey(20, accNoBalance.ScriptHash);
+            var entry = snapshot.GetAndChange(key, () => new StorageItem(new AccountState()));
+            entry.GetInteroperable<AccountState>().Balance = 1 * NativeContract.GAS.Factor;
+
+            key = NativeContract.GAS.CreateStorageKey(20, accBalance.ScriptHash);
+            entry = snapshot.GetAndChange(key, () => new StorageItem(new AccountState()));
+            entry.GetInteroperable<AccountState>().Balance = 100 * NativeContract.GAS.Factor;
+
+            snapshot.Commit();
+
+            // Make transaction
+
+            byte[] script;
+            using (ScriptBuilder sb = new())
+            {
+                sb.EmitDynamicCall(NativeContract.GAS.Hash, "refuel", accBalance.ScriptHash, 100 * NativeContract.GAS.Factor);
+                sb.Emit(OpCode.DROP);
+                sb.EmitSysCall(ApplicationEngine.System_Runtime_GasLeft);
+                script = sb.ToArray();
+            }
+
+            var signers = new Signer[]{ new Signer
+                {
+                    Account = accBalance.ScriptHash,
+                    Scopes =  WitnessScope.CalledByEntry
+                } ,
+                new Signer
+                {
+                    Account = accNoBalance.ScriptHash,
+                    Scopes =  WitnessScope.CalledByEntry
+                } };
+
+            var tx = wallet.MakeTransaction(snapshot, script, accBalance.ScriptHash, signers);
+            Assert.IsNotNull(tx);
+
+            // Check
+
+            using ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Application, tx, snapshot, settings: TestBlockchain.TheNeoSystem.Settings, gas: tx.NetworkFee);
+            engine.LoadScript(tx.Script);
+            Assert.AreEqual(VMState.HALT, engine.Execute());
+            Assert.AreEqual(1, engine.ResultStack.Count);
+            Assert.AreEqual(100_00300140, engine.ResultStack.Pop().GetInteger());
```

Comment:
Should this be precisely 100? I mean, if tx fee value is exact, and 100 is added up, we should perhaps rest with 100, right? Anyway, value is close to it.